apps/wallet/src/shared/utils/bip39.ts

```diff
@@ -12,7 +9,31 @@ import { wordlist } from '@scure/bip39/wordlists/english';
  * @returns a 12 words string separated by spaces.
  */
 export function generateMnemonic(): string {
-    return bip39GenerateMnemonic(wordlist);
+    return bip39.generateMnemonic(wordlist);
+}
+
+/**
+ * Converts mnemonic to entropy (byte array) using the english wordlist.
+ *
+ * @param mnemonic 12-24 words
+ *
+ * @return the entropy of the mnemonic (Uint8Array)
+ */
+
+export function mnemonicToEntropy(mnemonic: string): Uint8Array {
```

Comment:
Given the preliminary audit, we probably want to avoid the mnemonic hitting memory when we can. I think in this case, we can avoid it by just having a generateEntropy function exposed, and not using these conversions (we would only ever need to go from entropy -> mnemonic). The method to do so isn't directly exposed but is incredibly simple to re-implement in this bip39 file: https://github.com/paulmillr/scure-bip39/blob/main/src/index.ts#L42

Reply:
Ok I think I see what you mean, there is though 1-2 cases where we need to go from mnemonic to entropy. 1. Import wallet and 2. migrate the existing stored wallet. So probably we need this but I will change all the places to use only entropy (where possible)

Reply:
Yeah this makes sense to me. I think so long as we only keep it in memory during the execution of that piece of work (importing wallet, migrating) and immediately convert it to entropy and null out the raw mnemonic value, we should be okay.